src/test/java/com/coveo/spillway/functional/SpillwayFunctionalTests.java

```diff
@@ -101,14 +101,14 @@ public void oneMillionConcurrentRequestsWith100Threads() throws InterruptedExcep
     threadPool.shutdown();
     threadPool.awaitTermination(1, TimeUnit.MINUTES);
 
-    assertThat(counter.get()).isEqualTo(ONE_MILLION);
+    assertThat(counter.get()).isAtMost(ONE_MILLION + 10);
```

Comment:
Since we dropped the sync and increase the processing time of the spillway, we cant be sure of a precise result anymore.
The inner ConcurrentHashMap seemed to have helped and if we can optimize the Spillway check that would help too.